### PR TITLE
fix(browser): Moving to es6-shim broke most browsers

### LINKED ISSUE #4216: Moving to es6-shim broke most browsers
In fact, only Chrome and Edge made it through.

As far as I can see, 2 issues are causing this in both unit tests and examples:

1) Firefox: `Error: Expected int32 as second argument in c:/Dev/angular_mlaval/node_modules/es6-shim/es6-shim.js (line 1101)`
This is a known issue in es6-shim: https://github.com/paulmillr/es6-shim/issues/347
It comes from an actual bug in Firefox, which was apparently fixed in Firefox Nightly.

2) All other broken browsers: es6-shim doesn't polyfill Symbol and we have a few occurence of `Symbol.iterator` in the framework.
Internally, es6-shim uses a string if it doesn't exist, but doesn't expose it.
Issue opened for that: https://github.com/paulmillr/es6-shim/issues/360

### PR DESCRIPTION
**Pull request by marclaval**
Fixes #4216

For the 2 issues:
1) Thanks to advices in https://github.com/paulmillr/es6-shim/issues/347, loading es6-shim first solves the error.
2) Added a workaround to use the internal string from es6-shim when `Symbol.iterator`is not available.

### CODE REVIEW

**Comment by mlaval** [comment]
Benchmarks are broken and need to be fixed, but at least unit tests are now running normally (with real errors to fix ...)
@rkirov Can you have a look please?

File: modules/angular2/src/core/facade/lang.ts [hunk 1]
```diff
@@ -351,3 +351,24 @@ export function setValueOnPath(global: any, path: string, value: any) {
   }
   obj[parts.shift()] = value;
 }
+
+// TODO(mlaval): remove after https://github.com/paulmillr/es6-shim/issues/360
```
Comment (rkirov) [inline modules/angular2/src/core/facade/lang.ts]: Looks like issue #360 is closed and `Symbol.iterator` won't be exposed. I don't like that this ties us up to `es6-shim`, which means angular2 bundle can't be used with `core.js`. Change the comment to specify that this workaround is `es6-shim` specific.

File: tools/broccoli/trees/browser_tree.ts [hunk 2]
```diff
@@ -138,12 +138,12 @@ module.exports = function makeBrowserTree(options, destinationPath) {
 
   var vendorScriptsTree = flatten(new Funnel('.', {
     files: [
+      'node_modules/es6-shim/es6-shim.js',
```
Comment (rkirov) [inline tools/broccoli/trees/browser_tree.ts]: OOC, why does order matter?

File: modules/angular2/src/core/facade/lang.ts (hunk 1 above)
Comment (mlaval) [inline modules/angular2/src/core/facade/lang.ts]: `core-js` defines `Symbol.iterator`, so it shouldn't be a problem.

File: tools/broccoli/trees/browser_tree.ts (hunk 2 above)
Comment (mlaval) [inline tools/broccoli/trees/browser_tree.ts]: Not sure in fact. Looking at the comments in https://github.com/paulmillr/es6-shim/issues/347, it was something worth trying.
Anyway, this constraint will go away when the root bug will land in Firefox.

**Pull request closed without merging**